### PR TITLE
chore(core): Improve parse_wit_package_name handling

### PR DESCRIPTION
**Pull request by joonas**
## Feature or Problem

This fixes previously broken test cases, adds an actual test to run them.

### CODE REVIEW

File: crates/core/src/wit.rs
Comment (vados-cosmonic) [inline crates/core/src/wit.rs:308]: Did you by chance miss the doctests? I think the new cases here are `wasi:http` and `wasi:http@0.2.2` so if you could put them up there that would be great

**Review by vados-cosmonic — COMMENTED** [review]
LGTM :rocket: If we could keep the tests in the same place that would be great (and instruct people using it)

File: crates/core/src/wit.rs
Comment (joonas) [reply crates/core/src/wit.rs:308]: so tl;dr why I'm adding them here is:
1. `cargo nextest`, which is what we use for CI, [does not support running doctests](https://github.com/nextest-rs/nextest/issues/16).
2. `nixify`, which is what we use for running tests, didn't have the ability to run doctests until [support for it just landed recently](https://github.com/ipetkov/crane/pull/720), but since that hasn't yet made it to `nixify` (@rvolosatovs mentioned he was looking to add it next week), I figured it would be nice to run these.

Comment (vados-cosmonic) [reply crates/core/src/wit.rs:321]: Ah this is news to me, thanks for explaining!

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: crates/core/src/wit.rs
Comment (joonas) [reply crates/core/src/wit.rs:321]: No worries, I was a little surprised to find this out myself!

**Review by ricochet — APPROVED** [review]

**Pull request merged**
